test/Microsoft.AspNetCore.SignalR.Tests.Utils/PipeReaderExtensions.cs

```diff
@@ -28,8 +28,8 @@ public static class PipeReaderExtensions
                 }
                 finally
                 {
-                    // Consume nothing, just wait for everything
-                    pipeReader.AdvanceTo(result.Buffer.Start, result.Buffer.End);
+                    // Don't consume or advance
+                    pipeReader.AdvanceTo(result.Buffer.Start, result.Buffer.Start);
```

Comment:
Is this correct? Why did the behavior change?

Reply:
I suppose it doesn't really hurt the logic of this method though.

Reply:
Before, the next read would wait until another message, and not grab the one that this function read.

Reply:
Because it might leave pipe in the state where subsequent ReadAsync would never complete(no more data is written) and I doubt it desired behavior of WaitToRead method.